pkg/build/config.go

```diff
@@ -9,8 +9,14 @@ import (
 
 // GenerateConfigFromLabels generates the S2I Config struct from the Docker
 // image labels.
-func GenerateConfigFromLabels(config *api.Config) error {
-	result, err := docker.GetBuilderImage(config)
+func GenerateConfigFromLabels(config *api.Config, rebuild bool) error {
```

Comment:
The fact that `GenerateConfigFromLabels` method also pulling the image looks strange and unnatural. Because we're invoking this method only from twice, I suggest to pass `PullResult` to this method as an argument and pull the image from the method that invokes `GenerateConfigFromLabels`. In this case, we can also remove the `rebuild` argument.